Reflective access warning in Qute

We have these warnings in the build logs:

```
Error: -04 15:48:27,954 INFO  [io.quarkus] (main) Quarkus[ERROR] WARNING: An illegal reflective access operation has occurred
Error:  WARNING: Illegal reflective access by io.quarkus.qute.ReflectionValueResolver (file:/D:/a/quarkus/quarkus/independent-projects/qute/core/target/classes/) to method java.util.TreeMap$EntrySet.iterator()
Error:  WARNING: Please consider reporting this to the maintainers of io.quarkus.qute.ReflectionValueResolver
Error:  WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
Error:  WARNING: All illegal access operations will be denied in a future release
```

--- a/independent-projects/qute/core/src/main/java/io/quarkus/qute/ReflectionValueResolver.java
+++ b/independent-projects/qute/core/src/main/java/io/quarkus/qute/ReflectionValueResolver.java
@@ -4,6 +4,9 @@ import java.lang.reflect.Array;
 import java.lang.reflect.Field;
 import java.lang.reflect.Method;
 import java.lang.reflect.Modifier;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
 import java.util.Objects;
 import java.util.Optional;
 import java.util.concurrent.CompletableFuture;
@@ -100,25 +103,40 @@ public class ReflectionValueResolver implements ValueResolver {
         Method foundGetterMatch = null;
         Method foundBooleanMatch = null;
 
-        for (Method method : clazz.getMethods()) {
-            if (!isMethodValid(method)) {
-                continue;
-            }
-            if (method.isBridge()) {
-                continue;
-            }
-            if (name.equals(method.getName())) {
-                foundMatch = method;
-            } else if (matchesPrefix(name, method.getName(),
-                    GET_PREFIX)) {
-                foundGetterMatch = method;
-            } else if (isBoolean(method.getReturnType()) && (matchesPrefix(name, method.getName(),
-                    IS_PREFIX) || matchesPrefix(name, method.getName(), HAS_PREFIX))) {
-                foundBooleanMatch = method;
-            }
+        // Explore interface methods first...
+        List<Class<?>> classes = new ArrayList<>();
+        Collections.addAll(classes, clazz.getInterfaces());
+        Class<?> superClass = clazz.getSuperclass();
+        while (superClass != null) {
+            Collections.addAll(classes, superClass.getInterfaces());
+            superClass = superClass.getSuperclass();
         }
-        if (foundMatch == null) {
-            foundMatch = (foundGetterMatch != null ? foundGetterMatch : foundBooleanMatch);
+        classes.add(clazz);
+
+        for (Class<?> clazzToTest : classes) {
+            for (Method method : clazzToTest.getMethods()) {
+                if (!isMethodValid(method)) {
+                    continue;
+                }
+                if (method.isBridge()) {
+                    continue;
+                }
+                if (name.equals(method.getName())) {
+                    foundMatch = method;
+                } else if (matchesPrefix(name, method.getName(),
+                        GET_PREFIX)) {
+                    foundGetterMatch = method;
+                } else if (isBoolean(method.getReturnType()) && (matchesPrefix(name, method.getName(),
+                        IS_PREFIX) || matchesPrefix(name, method.getName(), HAS_PREFIX))) {
+                    foundBooleanMatch = method;
+                }
+            }
+            if (foundMatch == null) {
+                foundMatch = (foundGetterMatch != null ? foundGetterMatch : foundBooleanMatch);
+            }
+            if (foundMatch != null) {
+                break;
+            }
         }
         return foundMatch;
     }

--- a/independent-projects/qute/core/src/test/java/io/quarkus/qute/ReflectionResolverTest.java
+++ b/independent-projects/qute/core/src/test/java/io/quarkus/qute/ReflectionResolverTest.java
@@ -1,0 +1,20 @@
+package io.quarkus.qute;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.util.Map;
+import java.util.TreeMap;
+import org.junit.jupiter.api.Test;
+
+public class ReflectionResolverTest {
+
+    @Test
+    public void testReflectionResolver() {
+        Map<Integer, String> treeMap = new TreeMap<>(Integer::compare);
+        treeMap.put(2, "bar");
+        treeMap.put(1, "foo");
+        assertEquals("foo", Engine.builder().addDefaults().addValueResolver(new ReflectionValueResolver()).build()
+                .parse("{map.entrySet.iterator.next.value}").data("map", treeMap).render());
+    }
+
+}

